ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/SignedAggregateAndProof.java

```diff
@@ -27,9 +27,10 @@ public class SignedAggregateAndProof
   public static class SignedAggregateAndProofSchema
       extends ContainerSchema2<SignedAggregateAndProof, AggregateAndProof, SszSignature> {
 
-    public SignedAggregateAndProofSchema(final AggregateAndProofSchema aggregateAndProofSchema) {
+    public SignedAggregateAndProofSchema(
+        final AggregateAndProofSchema aggregateAndProofSchema, final boolean isElectra) {
       super(
-          "SignedAggregateAndProof",
+          isElectra ? "SignedAggregateAndProofElectra" : "SignedAggregateAndProofPhase0",
```

Comment:
i do wonder if we can get this information from the schema passed in rather than having a boolean flag which won't really scale out if we have to end up changing this in a future fork...

Reply:
Well, removed that boolean and used the attestation schema to determine that instead, it looks cleaner but yet all this schema versioning will need a refactoring in the future...

Reply:
yeh but by doing something like your new version, we no longer need 'isElectra', 'isF*', 'isG*' etc etc, so i think its closer to what we want...

Reply:
I guess you're talking about the other usages of the `isElectra` "hack", the problem is that in those other cases, we don't have access to the fork info from the available schemas. Take for example `IndexedAttestationSchema`, there's no way to know at which fork we're at without passing an additional info in the constructor. It played out well for the `AggregateAndProofSchema` and the `SignedAggregateAndProofSchema` because both have access to the attestation schema which can be used to determine if we reached Electra.
Some thoughts here:
- The really proper way to solve this is to extends the Electra affected schemas the same way we did for the `AttestationPhase0` and `AttestationElectra` but that means a huge refactoring and clearly must be done in a separate PR. We've already discussed this solution but thought we could do it later because the refactoring will impact the beacon block body schema...
- The idea is definitely not having `isF`, `isG`... that definitely doesn't scale and is almost an anti pattern. It's just a "hacky" intermediate solution